crates/starfish/core/src/commit_observer.rs

```diff
@@ -315,7 +324,60 @@ impl CommitObserver {
             .metrics
             .node_metrics
             .sub_dags_per_commit_count
-            .observe(committed.len() as f64);
+            .observe(pending_sub_dags.len() as f64);
+
+        // Now report transaction-related metrics for committed subdags
+        for commit in committed_sub_dags {
+            info!(
+                "Consensus available subdag {} with leader {} has transactions from {} blocks",
+                commit.commit_ref,
+                commit.leader,
+                commit.transactions.len()
+            );
+
+            // Report the actual number of committed transactions
+            metrics.transactions_per_commit_count.observe(
+                commit
+                    .transactions
+                    .iter()
+                    .map(|x| x.transactions().len())
+                    .sum::<usize>() as f64,
+            );
+
+            let block_refs_for_committed_txs = commit
+                .transactions
+                .iter()
+                .map(|tx| tx.block_ref())
+                .collect::<Vec<_>>();
+
+            // TODO: decide whether we want to read from storage here or not as it could
```

Comment:
I think we can ignore metrics for old transactions on a node that is experiencing some problems anyway  (networking or otherwise) and not add more IO requests.